### PR TITLE
ADAL to MSAL migration

### LINKED ISSUE #2782: Migrate to MSAL from adal-node
On inspection of https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/2023, there is now support for using client certificates in MSAL, which means we can continue moving away from adal-node.

@Zerryth, please reach out to either @carlosscastro or myself for any questions.

Tentatively placing in R11 milestone.

### PR DESCRIPTION
**Pull request by Zerryth**
Fixes #2782 

## Description
Migrate from ADAL to MSAL

**NOTE: `msal-node` is still in preview.**
- Won't be folding in these changes into R12

Quick draft PR, b/c I'm curious to know if msal-node's dependencies will break browser bot build and want to run ADO pipeline to test it. 

Only quickly swapped out token from ADAL res with token from MSAL res and further implementation cleanup necessary

### CODE REVIEW

File: libraries/botframework-connector/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,615 @@
+{
```
Comment (joshgummersall) [inline libraries/botframework-connector/package-lock.json:1]: These should be `gitignore`d - make sure you're using `yarn`.

**Comment by Zerryth** [comment]
Good news: the msal package we need to use is no longer preview and has no GA'd :)

Since this is stale, I'm just gonna start the PR over -- this was just a quick test anyways

**Comment by joshgummersall** [comment]
Awesome - thanks @Zerryth. If we can get this in this week that would be excellent, and if you need any assistance/reviews please ping me on Teams.

**Comment by Zerryth** [comment]
Resumed working on this actively now for the past couple of days, and unfortunately it doesn't look like it'll make it in this week, since it turns out not everything that uses ADAL is private (some are protected--like `AppCredentials.refreshingToken` is of type `Promise<adal.TokenResponse>`, for example, and the subclasses of `AppCredentials` base refreshing a token off it. *Sigh.* 😣 ). 

That makes it more complicated than just simply just ripping out ADAL pieces and replacing with MSAL, so I need to figure out a way to refactor this to be able to use MSAL and not break those who are  currently using ADAL

**Comment by joshgummersall** [comment]
All good! Thanks for the update 😄

**Pull request closed without merging**